edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/RouteConfiguration.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using Microsoft.Azure.Devices.Routing.Core;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    public class RouteConfiguration
+    {
+        public RouteConfiguration(string route)
+            : this(route, RouteFactory.DefaultPriority, 0)
+        {
+        }
+
+        [JsonConstructor]
+        public RouteConfiguration(string route, uint priority, uint timeToLiveSecs)
+        {
+            this.Route = route;
+            this.Priority = priority;
+            this.TimeToLiveSecs = timeToLiveSecs;
+        }
+
+        [JsonProperty(PropertyName = "route")]
+        public string Route { get; }
+
+        [DefaultValue(RouteFactory.DefaultPriority)]
+        [JsonProperty(PropertyName = "priority", DefaultValueHandling = DefaultValueHandling.Populate)]
+        public uint Priority { get; }
+
+        [JsonProperty(PropertyName = "timeToLiveSecs")]
+        public uint TimeToLiveSecs { get; }
+    }
+
+    public class RouteConfigurationDictionaryConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer) => throw new NotSupportedException();
+
+        public override object ReadJson(JsonReader reader, Type type, object value, JsonSerializer serializer)
+        {
+            var routes = new Dictionary<string, RouteConfiguration>();
+            JToken token = JToken.Load(reader);
+
+            foreach (JToken child in token.Children())
+            {
+                if (child.First.Type == JTokenType.String)
+                {
+                    // Legacy routes are just string properties
+                    routes.Add(child.First.Path, new RouteConfiguration(child.First.ToObject<string>()));

```

Comment:
make sure child.First.ToObject<string>() not return null or white spaces.

Reply:
Do you have tests for this?

Reply:
was going to add testing as part of the integration tests WI, right now I just verified that existing manifests aren't broken via E2E tests

---
In reply to: [376244854](https://github.com/Azure/iotedge/pull/2469#discussion_r376244854) [](ancestors = 376244854)

Reply:
I think this should be checked in the constructor of the concrete object, not here.

---
In reply to: [376169406](https://github.com/Azure/iotedge/pull/2469#discussion_r376169406) [](ancestors = 376169406)

Reply:
see other comments, I think it makes more sense to do validation check in the object-ctor.  Null checks shouldn't be necessary as we're expected to throw an exception in that case anyways.

---
In reply to: [376237140](https://github.com/Azure/iotedge/pull/2469#discussion_r376237140) [](ancestors = 376237140)

Reply:
I came to the realization that just verifying that existing behavior isn't broken already takes me half way there anyways, so I'll just add the other half and be done with integration tests.

---
In reply to: [377288342](https://github.com/Azure/iotedge/pull/2469#discussion_r377288342) [](ancestors = 377288342,376244854)